### PR TITLE
Get mutations to work with codegen

### LINKED ISSUE #2792: Query Codegen does not work for mutations.
## Describe the Bug

Query codegen seems to assume that it is only generating client objects for `query` objects and not `mutation` objects.  

The error seems to be in here:

https://github.com/strawberry-graphql/strawberry/blob/6cf8238c9cb11eda69a912e493c6a477b1d2a6be/strawberry/codegen/query_codegen.py#L258-L259

This assertion fails because the objects that are relevant are to be found via `self.schema.get_type_by_name("Mutation")` rather than `self.schema.get_type_by_name("Query")`.  A somewhat simple fix seems to be:

```py
query_type = self.schema.get_type_by_name(operation_definition.operation.value.title())
```

I have no idea if this is a proper fix for this bug, but later in the method, it seems like `mutation` and `subscription` are intended to be supported:

https://github.com/strawberry-graphql/strawberry/blob/6cf8238c9cb11eda69a912e493c6a477b1d2a6be/strawberry/codegen/query_codegen.py#L271-L274

## System Information

 - Operating system:
 - Strawberry version (if applicable):

### PR DESCRIPTION
**Pull request by mgilson**
## Description

Currently codegen does not work with `mutation`s.  The reason is because strawberry does not look in the correct location in the schema to find the associated types.  This PR introduces a regression test and a fix.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #2792 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Matt Gilson for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/codegen/query_codegen.py [hunk 1]
```diff
@@ -255,7 +255,9 @@ def _convert_directives(
     def _convert_operation(
         self, operation_definition: OperationDefinitionNode
     ) -> GraphQLOperation:
-        query_type = self.schema.get_type_by_name("Query")
+        query_type = self.schema.get_type_by_name(
+            operation_definition.operation.value.title()
+        )
```
Comment (patrick91) [inline strawberry/codegen/query_codegen.py:260]: ah, good catch! :D

**Review by patrick91 — APPROVED** [review]
This is amazing, well done!

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**